drivers/hmis/spec/system/hmis/form_hud_elements_spec.rb

```diff
@@ -0,0 +1,335 @@
+
+      expect(e1.income_benefits.sole.insurance_from_any_source).to eq(0)
+    end
+
+    it 'autoselects Yes when any health insurance provider is selected in table' do
+      find('[aria-label="Veteran\'s Health Administration (VHA)"]').check
+      find('[aria-label="Medicare"]').check
+      expect(mui_radio_value_for('Covered by Health Insurance')).to eq('YES')
+
+      expect do
+        save_ignoring_warnings
+      end.to change(e1.income_benefits, :count).by(1)
+
+      expect(e1.income_benefits.sole.insurance_from_any_source).to eq(1)
+      expect(e1.income_benefits.sole.vha_services).to eq(1)
+      expect(e1.income_benefits.sole.medicare).to eq(1)
+    end
+  end
+
+  describe 'disabilities' do
+    context 'with project type of ES and no funders' do
+      let!(:p1) { create :hmis_hud_project, data_source: ds1, organization: o1, user: u1, funders: [] }
+
+      it 'renders limited disability component, only collecting overall disabling condition' do
+        assert_text 'Disabling Condition'
+        assert_no_text 'The following questions help determine if there are additional housing services or benefits available for the client.'
+        expect(mui_select_value_for('Disabling Condition')).to eq('Data not collected')
+      end
+
+      it 'saves overall disabling condition as yes' do
+        mui_select('Yes', from: 'Disabling Condition')
+
+        expect do
+          save_ignoring_warnings
+        end.to change(e1, :disabling_condition).to(1)
+      end
+
+      it 'saves overall disabling condition as no' do
+        mui_select('No', from: 'Disabling Condition')
+
+        expect do
+          save_ignoring_warnings
+        end.to change(e1, :disabling_condition).to(0)
+      end
+    end
+
+    context 'with project that collects granular disability info' do
+      it 'renders full disability component' do
+        assert_text 'Overall Disabling Condition'
+        assert_text 'The following questions help determine if there are additional housing services or benefits available for the client.'
+        assert_text 'Physical Disability' # etc.
+        click_link 'Disability' # scrolls Disability into view
+        expect(mui_table_value_for(row: 'Overall Disabling Condition', column: 'Status')).to eq('Data not collected')
+      end
+
+      it 'updates overall Disabling Condition when you select Yes' do
+        mui_table_select('Yes', row: 'Physical Disability', column: 'Status')
+        mui_table_select('Yes', row: 'Physical Disability', column: 'Disabling Condition')
+        expect(mui_table_value_for(row: 'Overall Disabling Condition', column: 'Status')).to eq('Yes')
+
+        expect do
+          save_ignoring_warnings
+        end.to change(e1, :disabling_condition).to(1)
+        expect(e1.disabilities.where(disability_type: 5).sole.indefinite_and_impairs).to eq(1)
+      end
+
+      it 'updates overall Disabling Condition when you select an always-disabling condition' do
```

Comment:
So this is going to be kind of annoying, but, I would feel more comfortable if this test asserts on a few more things, for the sake of being confident in HUD compliance:

* When you mark Developmental Disability as `yes`, the overall Disabling Condition becomes Yes **and it is disabled**. (Its important per data dictionary that the user can't override it to No)
  * Then, when you clear or change Developmental Disability, overall Disabling Condition is no longer disabled and its cleared. (It's important that the field is interactable)
* If you open the assessment an enrollment that already has a value for Enrollment.DisablingCondition it should appear, and then if you add developmental disability it should auto change to Yes
* For the cases where you set just Physical Disability, adding an assertion to make sure Disabling Condition field is interactable (or add a step to set its value)
* Add a test case that fills in all the disability values and submits